test/tools/d_do_test.d

```diff
@@ -561,6 +561,10 @@ bool gatherTestParameters(ref TestArgs testArgs, string input_dir, string input_
         testArgs.compileOutput = readText(testArgs.compileOutputFile)
                                     .unifyNewLine() // Avoid CRLF issues
                                     .strip();
+
+        // Only sanitize separators from file types that support standalone \
+        if (!testArgs.compileOutputFile.endsWith(".json"))
+            testArgs.compileOutput = testArgs.compileOutput.unifyDirSep(envData.sep);
```

Comment:
Not sure if DMD should use `\\` in the JSON output but we can always remove this if the output is changed.